api/prometheus/v1/api.go

```diff
@@ -139,6 +165,38 @@ type SnapshotResult struct {
 	Name string `json:"name"`
 }
 
+// RulesResult contains the result from querying the rules endpoint.
+type RulesResult struct {
+	Groups []RuleGroup `json:"groups"`
+}
+
+type RuleGroup struct {
+	Name     string  `json:"name"`
+	File     string  `json:"file"`
+	Rules    []Rule  `json:"rules"`
```

Comment:
Since a Rule can be either `recording` or `alerting`, I'm not sure if we actually want to use a stronger type here with two fields `RecordingRules` and `AlertingRules`, or put everything in the same list to preserve the rule order like https://github.com/prometheus/prometheus/blob/master/web/api/v1/api.go#L721 does.

Reply:
I will let @simonpasquier 	and @mxinden comment on that.